### DISCUSSION TITLE
What is the correct way to configure CSP in Tauri when using CSS-in-JS libraries?

### DISCUSSION THREAD

**Discussion by billywatkins** [tauri-apps/tauri#8578] [Q&A] [answered]
In the release build, Tauri does not render anything. After a bit of Googling, I found the same [issue](https://github.com/tauri-apps/tauri/discussions/5723), but I am still confused about the right CSP configuration to set when using CSS-in-JS libraries like styled-components.

Right now, this is my CSP:
```"csp": "default-src 'self'"```

Error:
<img width="898" alt="Screenshot 2024-01-10 at 1 32 44 PM" src="https://github.com/tauri-apps/tauri/assets/124872003/f30e0484-27db-4705-a194-748988267b62">

**Comment by billywatkins** [comment, marked as answer]
Okay, I finally ended up with this configuration in the tauri.conf.json file. Is this the right way to do it? I don't know. I'm leaving my comment here so people who come here may find this useful.

```json
  "security": {
      "dangerousDisableAssetCspModification": ["style-src"],
      "csp": "default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' asset:"
    }
```

EDIT:
> Based on my own experience, I've never seen an attack that leveraged the injection of CSS to cause harm. - https://scotthelme.co.uk/can-you-get-pwned-with-css/

I guess "unsafe-inline" is okay.

**Reply by FabianLars** [reply]
If your css-in-js library has no static export then this is indeed the only solution.

**Reply by billywatkins** [reply]
Thanks for clarifying it!

I checked both styled-components and Emotion; it seems that neither supports exporting CSS into separate files[1][2], I might be wrong.

1. emotion: https://stackoverflow.com/questions/58932468/is-it-possible-to-generate-css-files-using-emotion
2. styled-components: https://github.com/styled-components/styled-components/issues/1018#issuecomment-316827497
